### DISCUSSION TITLE
Save last workspace after shutdown instead of startup?

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse-platform/eclipse.platform#1400] [Ideas]
Following is the current behavior:

1. I start Eclipse with  Workspace `A`
2. I start Eclipse again, it says I have to use another workspace and I choose Workspace `B`
3. Now I close the Eclipse from (2)
4. Now I close/restart the Eclipse from (1)
5. Eclipse now is uses Workspace `B` as the last / default Workspace

I find this behavior a bit counter intuitive especially for the restart case, but as it is working that way "forever" I'd like to discuss if its only me or if others agree that one can expect to use the workspace of the _last closed_ Eclipse instance in contrast as the _last opened_ Eclipse instance.

**Comment by merks** [comment]
Indeed between 1 and 4 could be several days, and one might have long forgotten having done 2 and 3.    Moreover, it seems reasonable to expect that a restart would always reuse the current workspace.  Perhaps it should save on both startup and shutdown; I have been known to kill and IDE that's hung...

**Reply by HannesWell** [reply]
From my experience, the behavior Christoph described is indeed often the one I want but instead what I intend is what he suggests.
Bug saving it on both shutdown and start-up sounds even better to me.

**Reply by fedejeanne** [reply]
Expanding on step nr 4., I'd expect that...
1. A restart (`File > Restart`) uses the **current** WS _i.e._ `A`.
2. A new start (`File > Exit` + double-click on `eclipse.exe`) offers/uses the **last opened** WS _i.e._ `B`. I guess that would mean saving the WS during start-up. EDIT: if the option "Use this one and do not ask again" is in play, the **last preferred (saved during startup)** WS should be used.

**Reply by laeubi** [reply]
Yes saving at start **and** shutdown (new behavior) versus save at start only (current behavior).
